src/SIUnits.jl

```diff
@@ -7,15 +7,28 @@ module SIUnits
     import Base: ==, +, -, *, /, .+, .-, .*, ./, //, ^
     import Base: promote_rule, promote_type, convert, show, mod
 
-    immutable SIQuantity{T<:Number,m,kg,s,A,K,mol,cd,rad,sr} <: Number
+    typealias UnitTuple NTuple{9,Int64}
```

Comment:
There is a test case that checks that `sqrt(1s)` throws an error, so I assume that hasn't worked for a while.

But with some minor changes,  I was able to switch it to
```julia
typeaias UnitTuple NTuple{9,Rational{Int64}}
```
It seems to work fine with some basic testing, and I would think this would be superset of the current functionality of this PR.  Not sure if this is what you had in mind, or something more generic like
```julia
typeaias UnitTuple{N<:Number} NTuple{9,N}
```
I would guess that this would require a decent amount of work to get working